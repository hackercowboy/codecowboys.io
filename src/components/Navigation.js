/* eslint-disable react/no-array-index-key */
import React, { useState, useMemo } from 'react';
import PropTypes from 'prop-types';
import { Events } from 'react-scroll';

import Hamburger from './Hamburger';

function Navigation({ children }) {
  const [open, setOpen] = useState(false);

  useMemo(() => {
    Events.scrollEvent.register('begin', () => {
      setOpen(false);
    });
  });

  return (
    <nav>
      <Hamburger open={open} onClick={() => setOpen(!open)} />
      <div className={open ? 'navigation-items open' : 'navigation-items'}>
        { children}
      </div>
    </nav>
  );
}

Navigation.propTypes = {
  children: PropTypes.node,
};

Navigation.defaultProps = {
  children: undefined,
};

export default Navigation;
