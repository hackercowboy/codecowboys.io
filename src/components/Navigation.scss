@import '../styles/colors';

nav {
  display: flex;
  justify-content: flex-end;
  width: 100%;

  .navigation-items {
    display: flex;
  }
}

@media (max-width : 576px) {
  nav {
    .navigation-items {
      background: rgba(255, 255, 255, 0.9);
      border-bottom: solid 1px $light-color;
      display: flex;
      flex-direction: column;
      left: 0;
      padding: 10px;
      position: absolute;
      top: 58px;
      transform: scaleY(0);
      transform-origin: left top;
      transition: all 0.2s ease-in-out;
      width: 100%;
    }

    .navigation-items.open {
      
      transform: scaleY(1);
    }  
  }
}


