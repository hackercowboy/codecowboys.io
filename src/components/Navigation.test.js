import React from 'react';
import { shallow } from 'enzyme';
import { Events } from 'react-scroll';

import Navigation from './Navigation';

jest.mock('react-scroll', () => ({ Events: { scrollEvent: { register: jest.fn() } } }));

describe('<Navigation/>', () => {
  it('should render', () => {
    const wrapper = shallow(<Navigation />);
    expect(wrapper.debug()).toMatchSnapshot();

    wrapper.find('Hamburger').props().onClick();
    expect(wrapper.debug()).toMatchSnapshot();

    expect(Events.scrollEvent.register).toBeCalled();
    Events.scrollEvent.register.mock.calls[0][1]();
    wrapper.update();

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
