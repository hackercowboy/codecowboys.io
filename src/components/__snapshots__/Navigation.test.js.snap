// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> should render 1`] = `
"<nav>
  <Hamburger open={false} onClick={[Function: onClick]} />
  <div className=\\"navigation-items\\" />
</nav>"
`;
